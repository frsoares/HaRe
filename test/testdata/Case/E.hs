module E where
-- Test for refactor of if to case in a complex sub level
-- The comments on the then and else legs should be preserved

foo x = bob x f
  where
    f = (if (odd x)
          then do
            -- This is an odd result
            bob x 1
          else do
            -- This is an even result
            bob x 2) + 2

bob x y = x + y

