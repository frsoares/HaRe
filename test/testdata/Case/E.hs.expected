module E where
-- Test for refactor of if to case in a complex sub level
-- The comments on the then and else legs should be preserved

foo x = bob x f
  where
    f = (case (odd x) of
             True  -> do
                   -- This is an odd result
                   bob x 1
             False -> do
                   -- This is an even result
                   bob x 2

                   ) + 2

bob x y = x + y

